src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2324,7 +2325,26 @@ because xbuild doesn't support framework reference assemblies.
   </ItemGroup>
 </Target>
 
-<Target Name="_GetAddOnPlatformLibraries" DependsOnTargets="_GenerateJavaStubs">
+<Target Name="_SetupEmbeddedDSOs"
+    Condition=" '$(_EmbeddedDSOsEnabled)' == 'True' "
+    Inputs="$(IntermediateOutputPath)android\AndroidManifest.xml"
+    Outputs="$(IntermediateOutputPath)dsoenvironment.txt">
+  <WriteLinesToFile
+      File="$(IntermediateOutputPath)dsoenvironment.txt"
+      Lines="__XA_DSO_IN_APK=1"
+      Overwrite="True"
+      WriteOnlyWhenDifferent="True"
+  />
+  <PropertyGroup>
+    <AndroidStoreUncompressedFileExtensions>.so;$(AndroidStoreUncompressedFileExtensions)</AndroidStoreUncompressedFileExtensions>
```

Comment:
Won't that fail if `AndroidStoreUncompressedFileExtensions` is set on command line with `/p:AndroidStoreUncompressedFileExtensions=something`? Doesn't `msbuild` mark such a propery R/O for the rest of the session?

Reply:
I'm not sure if I've seen that behavior. Maybe it was an xbuild thing?

The new test I added, it still passes if I change it to:
```csharp
Assert.IsTrue (b.Build (proj, parameters: new [] { "AndroidStoreUncompressedFileExtensions=png" }), "Build should have succeeded");
```
Then the binlog looks like it's doing the right thing:
![image](https://user-images.githubusercontent.com/840039/50641970-5ecec700-0f2f-11e9-859d-0fb538f9aa1c.png)

Reply:
@grendello asked:

> Won't that fail if `AndroidStoreUncompressedFileExtensions` is set on command line

No.  Command-line specified properties being read-only is an `xbuild`-ism.  Consider `X.targets`:

```xml
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <Foo>Default</Foo>
  </PropertyGroup>
  <Target Name="Default">
    <PropertyGroup>
      <Foo>$(Foo);Bar</Foo>
    </PropertyGroup>
    <Message Text="Foo=$(Foo)" Importance="High"= />
  </Target>
</Project>
```

Then run, overriding the `$(Foo)` property, comparing `msbuild` to `xbuild`:

```
$ msbuild /v:minimal /nologo /t:Default X.targets /p:Foo=Baz
  Foo=Baz;Bar
$ xbuild /v:minimal /nologo /t:Default X.targets /p:Foo=Baz
	Foo=Baz
```

Notice that with `msbuild` the Target-local `$(Foo)` override/update occurs, causing `Baz;Bar` to be printed; `$(Foo)` is *not* readonly in `msbuild`.  In `xbuild`, `$(Foo)` *is* readonly, which is why it prints out only `Baz`.

All Hail The Deprecation Of `xbuild`!